src/containers/ManchesterTramStatus/Tram/Tram.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { Component } from 'react';
+import './Tram.css';
+import axios from 'axios';
+import Widget from '../../../components/Widget';
+import StatusCard from '../StatusCard/StatusCard';
+
+const trams = [
+  { id: 'altrincham', name: 'Altrincham – Piccadilly' },
+  { id: 'ashton-under-lyne', name: 'Ashton-under-Lyne – MediaCityUK' },
+  { id: 'bury', name: 'Altrincham – Bury' },
+  { id: 'east didsbury', name: 'East Didsbury – Shaw and Crompton' },
+  { id: 'eccles via mediacityuk', name: 'Eccles - MediaCityUK' },
+  { id: 'manchester airport', name: 'Manchester Airport – Victoria' },
+  { id: 'mediacityuk', name: 'Ashton-under-Lyne – Eccles' },
+  { id: 'rochdale via oldham', name: 'East Didsbury – Rochdale Town Centre' }
+];
+
+/**
+ *  Create an initial array of tram statuses containing:
+ *  { key:ashton-under-lyne, line: lineName, status: 'Good Service', reason: null }
+ * @param list of tram objects
+ */
+function createStatuses(list) {
+  return list.map(item => {
+    return {
+      key: item.id,
+      line: item.name,
+      status: 'Good Service',
+      reason: null
+    };
+  });
+}
+
+/**
+ * Removes the last 3 characters of the date,
+ * removes seconds from the date received.
+ */
+function formatDate(date) {
+  return date.slice(0, -3);
+}
+
+class Tram extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      statuses: [],
+      lastUpdatedTime: ''
+    };
+
+    this.getData = this.getData.bind(this);
+    this.getData();
+    this.setGetDataInterval();
+  }
+
+  setGetDataInterval() {
+    const now = new Date();
+    const tenMinutes = 60000 * 10;
+    const thirtyMinutes = 60000 * 30;
+    if (now.getHours() >= 16 && now.getHours() < 19) {
+      this.interval = setInterval(this.getData, tenMinutes);
+    } else {
+      this.interval = setInterval(this.getData, thirtyMinutes);
```

Comment:
Correct me if I'm wrong, won't this just add a 30 mins interval on top of the 10 mins interval? Should you not clear it first?

Reply:
To be honest i don't think it would have worked either unless the app is set to refresh between those hours so I've updated to use componentDidMount and componentDidUpdate.
Clearing the interval there and then resetting.